tensorflow/python/ops/array_ops.py

```diff
@@ -1273,6 +1277,57 @@ def transpose(a, perm=None, name="transpose"):
     return ret
 
 
+def framesig(sig, frame_len, frame_step, parallel=False, name="framesig"):
+  """Frame a signal into overlapping frames.
+  May be used in front of spectral functions.
+  
+  For example:
+  
+  ```python
+  pcm = tf.placeholder(tf.float32, [None, 9152])
+  frames = tf.framesig(pcm, 512, 180)
+  magspec = tf.abs(tf.spectral.rfft(frames, tf.constant(512, shape=[1])))
+  image = tf.reshape(magspec, [-1, 49, 257, 1])
+  ```
+  
+  Args:
+    sig: A `Tensor` with `rank = 2`.
+    frame_len: The length of each frame.
+    frame_step: The step between frames.
+    parallel: Whether to use `tf.parallel_stack` or `tf.stack`
+    name: A name for the operation (optional).
+  
+  Returns:
+    A `Tensor` of frames with shape [batch_size, num_frames, frame_len].
+  """
+  sig_shape = sig.get_shape()
+  sig_rank = len(sig_shape)
+  
+  if sig_rank != 2:
+    raise ValueError("expected sig to have rank 2 but was " + sig_rank)
+  
+  sig_len = int(sig_shape[1])
+  
+  num_frames = 1 + int(math.ceil((1. * sig_len - frame_len) / frame_step))
+  pad_len = int((num_frames - 1) * frame_step + frame_len)
+  
+  with ops.name_scope(name, "framesig", [sig]) as name:
+    if pad_len != sig_len:
+      pad_sig = ops.pad(sig, [[0, 0], [0, pad_len - sig_len]], name=name)
+    else:
+      pad_sig = sig
+    
+    frames = []
+    
+    for i in xrange(num_frames):
```

Comment:
can you do this without a python for loop?  again, this requires the num_frames be known at graph build time which you should not assume.